From 0d418b4c3f71a3bf259e43b5236e702636a32ec4 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Mon, 23 May 2011 14:29:26 +0800
Subject: [PATCH 301/678] ENGR00143837 mx50 rd3: support android boot from RD3
 NAND device

Enable NAND gpio, recovery mode detect after boot from spi nor.
Change default env for loading kernel and uramdisk from NAND,
disabling elcdif lcd driver to support EPDC eink panel as default.
Enable recovery mode support and NAND/UBI/UBIFS command.

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 board/freescale/mx50_rdp/mx50_rdp.c |    6 ++----
 include/configs/mx50_rd3_android.h  |   13 +++++--------
 2 files changed, 7 insertions(+), 12 deletions(-)

diff --git a/board/freescale/mx50_rdp/mx50_rdp.c b/board/freescale/mx50_rdp/mx50_rdp.c
index 3bf1bdd..8470af0 100644
--- a/board/freescale/mx50_rdp/mx50_rdp.c
+++ b/board/freescale/mx50_rdp/mx50_rdp.c
@@ -510,6 +510,7 @@ void setup_gpmi_nand(void)
 
 		break;
 	case 1:
+	case 3:
 		if ((src_sbmr & 0x00000004) >> 2)
 			*(u32 *)(IOMUXC_BASE_ADDR + PAD_GRP_START + 0xc) =
 								(0x1 << 13);
@@ -1410,6 +1411,7 @@ int check_recovery_cmd_file(void)
 		}
 		break;
 	case NAND_BOOT:
+	case SPI_NOR_BOOT:
 		env = getenv("mtdparts");
 		if (!env)
 			setenv("mtdparts", MTDPARTS_DEFAULT);
@@ -1426,10 +1428,6 @@ int check_recovery_cmd_file(void)
 				CONFIG_ANDROID_UBIFS_PARTITION_NM,
 				CONFIG_ANDROID_CACHE_PARTITION_NAND,
 				CONFIG_ANDROID_RECOVERY_CMD_FILE);
-		break;
-	case SPI_NOR_BOOT:
-		return 0;
-		break;
 	case UNKNOWN_BOOT:
 	default:
 		return 0;
diff --git a/include/configs/mx50_rd3_android.h b/include/configs/mx50_rd3_android.h
index 3d39584..8bb5189 100644
--- a/include/configs/mx50_rd3_android.h
+++ b/include/configs/mx50_rd3_android.h
@@ -125,13 +125,12 @@
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootcmd=run bootcmd_nand \0"				\
 		"bootcmd_nand=run bootargs_base bootargs_android;"	\
-		     "sf probe 1;"	\
-		     "sf read ${loadaddr} 0x40000 0x380000;"	\
-		     "sf read ${rd_loadaddr} 0x3C0000 0x40000;"	\
+		     "nand read ${loadaddr} 0x1400000 0x400000;"	\
+		     "nand read ${rd_loadaddr} 0x1900000 0x32000;"	\
 		     "bootm ${loadaddr} ${rd_loadaddr}\0"		\
 		"bootargs_android=setenv bootargs ${bootargs}  "	\
 		     "androidboot.console=ttymxc0 init=/init "		\
-		     "keypad gpmi:nand ubi.mtd=4\0"			\
+		     "keypad gpmi:nand ubi.mtd=4 video=mxc_elcdif_fb:off\0"	\
 		"bootcmd_android_recovery=run bootargs_base"		\
 		     " bootargs_android_recovery;"			\
 		     "nand read ${loadaddr} 0x1400000 0x400000; bootm\0"	\
@@ -238,7 +237,7 @@
 /*
  * GPMI Nand Configs
  */
-#undef CONFIG_CMD_NAND
+#define CONFIG_CMD_NAND
 
 #ifdef CONFIG_CMD_NAND
 	#define CONFIG_NAND_GPMI
@@ -299,7 +298,7 @@
 #elif defined(CONFIG_FSL_ENV_IN_SF)
 	#define CONFIG_ENV_IS_IN_SPI_FLASH	1
 	#define CONFIG_ENV_SPI_CS		1
-	#define CONFIG_ENV_OFFSET       (192 * 1024)
+	#define CONFIG_ENV_OFFSET       (768 * 1024)
 #else
 	#define CONFIG_ENV_IS_NOWHERE	1
 #endif
@@ -307,7 +306,6 @@
 /*
  * Android support Configs
  */
-/*
 #define CONFIG_ANDROID_RECOVERY
 #define CONFIG_ANDROID_RECOVERY_BOOTARGS_MMC \
        "setenv bootargs ${bootargs} init=/init root=/dev/mmcblk0p4"    \
@@ -333,7 +331,6 @@
 #define MTD_ACTIVE_PART "nand0,4"
 #define CONFIG_RBTREE
 #define CONFIG_LZO
-*/
 
 #define CONFIG_ANDROID_SYSTEM_PARTITION_MMC 2
 #define CONFIG_ANDROID_RECOVERY_PARTITION_MMC 4
-- 
1.7.7.4

